pkg/kubectl/resource_printer.go

```diff
@@ -1174,3 +1184,26 @@ func indirect(v reflect.Value) (rv reflect.Value, isNil bool) {
 	}
 	return v, false
 }
+
+// JSONPathPrinter is an implementation of ResourcePrinter which formats data with jsonpath expression.
+type JSONPathPrinter struct {
+	rawTemplate string
+	template    *jsonpath.JSONPath
+}
+
+func NewJSONPathPrinter(tmpl []byte) (*JSONPathPrinter, error) {
```

Comment:
It's weird that you are casting to []byte above, and then back to string below, just make the type of `tmpl` be `string`